game-app/game-core/src/main/java/games/strategy/triplea/ai/pro/ProCombatMoveAi.java

```diff
@@ -1729,21 +1754,66 @@ private Map<Unit, Set<Territory>> tryToAttackTerritories(
         }
       }
       if (minWinTerritory != null) {
-        attackMap.get(minWinTerritory).getBombardTerritoryMap().put(u, minBombardFromTerritory);
+        attackMap.get(minWinTerritory).getBombardTerritoryMap().put(unit, minBombardFromTerritory);
         attackMap.get(minWinTerritory).setBattleResult(null);
-        sortedUnitAttackOptions.remove(u);
+        sortedUnitAttackOptions.remove(unit);
         ProLogger.trace(
             "Adding bombard to "
                 + minWinTerritory
                 + ", units="
-                + u
+                + unit
                 + ", bombardFrom="
                 + minBombardFromTerritory);
       }
     }
     return sortedUnitAttackOptions;
   }
 
+  /** Data storage class for territories that could be bombed. */
+  static class BombedTerritoryData {
+    final Map<Unit, Pair<Integer, Integer>> minNeededAndMaxDamageMap = new HashMap<>();
+    @Getter int noAaBombingDefense;
+
+    public BombedTerritoryData() {
+      this.noAaBombingDefense = 1;
+    }
+
+    public void addPotentialTargets(List<Unit> targetUnits) {
+      for (Unit targetUnit : targetUnits) {
+        minNeededAndMaxDamageMap.compute(
+            targetUnit,
+            (targetUnitOld, damagePairOld) -> {
+              Integer unitMinDamageNeeded = (targetUnitOld == null ? 0 : damagePairOld.getFirst());
```

Comment:
Nit: I think it's a bit clearer if you check `damagePairOld == null`, even though it's semantically equivalent.

Also, let's rename these to `existingUnitKey` and `existingDamagePair` to make it a bit clearer the semantics of `compute()` since not everyone may be familiar with it.

Reply:
Fixed in [7e15c7a](https://github.com/frigoref/triplea/commit/7e15c7a7310f379b14fea38cc8836059d9d3e6e1).